include/assert.hpp

```diff
@@ -19,8 +19,7 @@
 
 #ifdef ASSERT_USE_MAGIC_ENUM
  // this is a temporary hack to make testing thing in compiler explorer quicker (it disallows simple relative includes)
- #include\
- "../third_party/magic_enum.hpp"
+ #include "magic_enum.hpp"
```

Comment:
Let's see, is this needed for the cmake setup? To get the makefile builds working we can probably just add `third_party/` as an include path.

Reply:
It's not, I put it back the way it was. 

The install layout:
```
.
├── include
│   └── assert
│       ├── assert // clients include this directory (#include <assert.hpp>)
│       │   └── assert.hpp
│       └── third_party
│           └── magic_enum.hpp
└── lib
    ├── cmake
    │   └── assert
    │       ├── assert-config.cmake
    │       ├── assert-config-version.cmake
    │       ├── assert_targets.cmake
    │       └── assert_targets-noconfig.cmake
    └── libassert.a
``` 
It's not ideal to have a third_party directory hanging in there but it's now nested under `assert` which should avoid issues.

Reply:
Thanks so much, I'll go ahead and merge the current changes. In the future would it be better to try to bundle assert.hpp and magic_enum.hpp into a single header file?

Reply:
I personally would not. Having a relative include directive in an exported header is somewhat annoying because you have to consider all contexts (install_interface + build_interface) and how these properties impact other targets transitively. But this case is fairly simple to handle. Some would probably argue that magic_enum.hpp should be in include because it is part of your public interface. I don't know if I agree.

That being said, this is what package managers are for. Unless you have made modifications that you can't upstream, you should not need to have third party code in your build-tree and/or install-tree. I'd use conan and then find_package(magic_enum) and call it a day. I assume you want to make libassert available through conan anyway at some point. 

I don't know why you need it, but I would consider getting rid of the makefile in favor of CMake. It's better to have one source of truth.